stdlib/mexpr/eq.mc

```diff
@@ -425,19 +455,38 @@ lang BoolPatEq = BoolPat
     else None ()
 end
 
-lang AndPatEq
+lang AndPatEq = AndPat
   sem eqPat (env : Env) (free : Env) (patEnv : NameEnv) (lhs : Pat) =
-  -- TODO(dlunde,2020-09-29)
+  | PAnd {lpat = l2, rpat = r2} ->
+    -- NOTE(gcaylak,2020-10-02): This check assumes order is important.
+    -- So 'l1 & r1 != l2 & r2' in terms of pattern matching.
+    -- Should we change this with irrelevant order?
+    match lhs with PAnd {lpat = l1, rpat = r1} then
+      match eqPat env free patEnv l1 l2 with Some (free,patEnv) then
+        eqPat env free patEnv r1 r2
+      else None ()
+    else None ()
 end
 
-lang OrPatEq
+lang OrPatEq = OrPat
   sem eqPat (env : Env) (free : Env) (patEnv : NameEnv) (lhs : Pat) =
-  -- TODO(dlunde,2020-09-29)
+  | POr {lpat = l2, rpat = r2} ->
+    -- NOTE(gcaylak,2020-10-02): This check assumes order is important.
+    -- So 'l1 | r1 != l2 | r2' in terms of pattern matching.
+    -- Should we change this with irrelevant order?
```

Comment:
This check is just for alpha equivalence, so having the order matter is correct, since the only thing we're supposed to be flexible in is naming. There will be another separate analysis down the line that can check for pattern equivalence